File: src/main/cpp/example_plugin/MyLayoutAnalysis.cpp

#include "MyLayoutAnalysis.h"
#include "../Image.h"

#include <iostream>

namespace transkribus {


MyLayoutAnalysis::MyLayoutAnalysis() {}

MyLayoutAnalysis::~MyLayoutAnalysis()
{
	std::cout << "MyLayoutAnalysis is deleted!" << std::endl;
}

void MyLayoutAnalysis::process(Image& image, const std::string& xmlInOut, const std::vector<std::string>& ids, const std::vector<std::string>& props) {
	std::cout << "MyLayoutAnalysis, process, xmlInOut =  " << xmlInOut << std::endl;
	std::cout << image << std::endl;
}

const std::string MyLayoutAnalysis::usage() const {
	return std::string("MyUsage");
}

const std::string MyLayoutAnalysis::getToolName() const {
	return std::string("MyToolName");
}

const std::string MyLayoutAnalysis::getVersion() const {
	return std::string("MyVersion");
}

const std::string MyLayoutAnalysis::getProvider() const {
	return std::string("MyProvider");
}


IModule * MyLayoutAnalysisModuleFactory::create(const std::vector<std::string>& pars)
{
	std::cout << "creating MyLayoutAnalysis" << std::endl;
	return new MyLayoutAnalysis;
}

}

File: src/main/java/eu/transkribus/interfaces/example_plugin/MyLayoutAnalysis.java
package eu.transkribus.interfaces.example_plugin;

import eu.transkribus.interfaces.ILayoutAnalysis;
import eu.transkribus.interfaces.types.Image;

public class MyLayoutAnalysis implements ILayoutAnalysis {

	@Override public String usage() {
		return "myUsage";
	}

	@Override public String getToolName() {
		return "myToolName";
	}

	@Override public String getVersion() {
		return "myVersion";
	}

	@Override public String getProvider() {
		return "myProvider";
	}

	@Override public void process(Image image, String xmlInOut, String[] ids, String[] props) {
		System.out.println("process of MyLayoutAnalysis, xmlInOut = "+xmlInOut);
	}

}


